internal/controllers/device_data_controller.go

```diff
@@ -290,26 +302,63 @@ func (udc *UserDevicesController) QueryDeviceErrorCodes(c *fiber.Ctx) error {
 		return helpers.GrpcErrorToFiber(err, "deviceDefSvc error getting definition id: "+ud.DeviceDefinitionID)
 	}
 
-	req := &QueryDeviceErrorCodesReq{}
-	if err := c.BodyParser(req); err != nil {
-		return fiber.NewError(fiber.StatusBadRequest, err.Error())
-	}
+	oi := services.NewOpenAI(*udc.Settings)
+	chtResp, err := oi.QueryDeviceErrorCodes(dd.Type.Make, dd.Type.Model, dd.Type.Year, req.ErrorCodes)
 	if err != nil {
-		return fiber.NewError(fiber.StatusBadRequest, err.Error())
+		return fiber.NewError(fiber.StatusInternalServerError, err.Error())
 	}
 
-	oi := services.NewOpenAI(*udc.Settings)
+	q := &models.ErrorCodeQuery{ID: ksuid.New().String(), UserDeviceID: udi, ErrorCodes: req.ErrorCodes, QueryResponse: chtResp}
+	err = q.Insert(c.Context(), udc.DBS().Writer, boil.Infer())
 
-	chtResp, err := oi.QueryDeviceErrorCodes(dd.Type.Make, dd.Type.Model, dd.Type.Year, req.ErrorCodes)
 	if err != nil {
-		return fiber.NewError(fiber.StatusInternalServerError, err.Error())
+		// TODO - should we return an error for this or just log it
+		udc.log.Err(err).Msg("Could not save user query response")
 	}
 
 	return c.JSON(&QueryDeviceErrorCodesResponse{
 		Message: chtResp,
 	})
 }
 
+// GetUserDevicesErrorCodeQueries godoc
+// @Description Returns all error codes queries for user devices
+// @Tags        user-devices
+// @Success     200 {object} controllers.GetUserDevicesErrorCodeQueriesResponse
+// @Security    BearerAuth
+// @Router      /user/devices/error-codes [get]
+func (udc *UserDevicesController) GetUserDevicesErrorCodeQueries(c *fiber.Ctx) error {
+	userID := helpers.GetUserID(c)
+
+	userDevices, err := models.UserDevices(
+		models.UserDeviceWhere.UserID.EQ(userID),
+		qm.Load(models.UserDeviceRels.ErrorCodeQueries),
+	).All(c.Context(), udc.DBS().Reader)
+
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return fiber.NewError(fiber.StatusNotFound, err.Error())
+		}
+		return err
+	}
+
+	resp := map[string][]GetUserDevicesErrorCodeQueriesResponse{}
+
+	for _, userDevice := range userDevices {
+		ud := []GetUserDevicesErrorCodeQueriesResponse{}
+		for _, erc := range userDevice.R.ErrorCodeQueries {
+			ud = append(ud, GetUserDevicesErrorCodeQueriesResponse{
+				Codes:       erc.ErrorCodes,
+				Description: erc.QueryResponse,
+			})
+		}
+
+		resp[userDevice.ID] = ud
+	}
+
+	return c.JSON(resp)
```

Comment:
I forgot something important: I would not return a map like this at the top level. I think it constrains what you can add to the output later too much.